src/algorithms/synthesis/syrec_synthesis.cpp

```diff
@@ -0,0 +1,1447 @@
+            append_circuit(_circ, *(get(boost::vertex_name, cct_man.tree)[current].circ), get(boost::vertex_name, cct_man.tree)[current].controls);
+            return true;
+        }
+        // assemble optimized circuits of successors
+        for (auto edge_it = out_edges(current, cct_man.tree).first; edge_it != out_edges(current, cct_man.tree).second; ++edge_it) {
+            if (!assemble_circuit(target(*edge_it, cct_man.tree))) return false;
+        }
+        return true;
+    }
+
+    bool standard_syrec_synthesizer::get_variables(const applications::variable_access::ptr& var, std::vector<unsigned>& lines) {
+        unsigned offset = _var_lines[var->var()];
+
+        if (!var->indexes().empty()) {
+            // change offset
+
+            // check if it is all numeric_expressions
+            unsigned n = var->var()->dimensions().size(); // dimensions
+            if ((unsigned)std::count_if(var->indexes().cbegin(), var->indexes().cend(), [&](const auto& p) { return dynamic_cast<applications::numeric_expression*>(p.get()); }) == n) {
+                for (unsigned i = 0u; i < n; ++i) {
+                    offset += dynamic_cast<applications::numeric_expression*>(var->indexes().at(i).get())->value()->evaluate(loop_map) *
+                              std::accumulate(var->var()->dimensions().begin() + i + 1u, var->var()->dimensions().end(), 1u, std::multiplies<>()) *
+                              var->var()->bitwidth();
+                }
+            } else {
+            }
+        }
+
+        if (var->range()) {
+            auto [nfirst, nsecond] = *var->range();
+
+            unsigned first  = nfirst->evaluate(loop_map);
+            unsigned second = nsecond->evaluate(loop_map);
+
+            if (first < second) {
+                for (unsigned i = first; i <= second; ++i) {
+                    lines.emplace_back(offset + i);
+                }
+            } else {
+                for (int i = (int)first; i >= (int)second; --i) {
+                    lines.emplace_back(offset + i);
+                }
+            }
+        } else {
+            for (unsigned i = 0u; i < var->var()->bitwidth(); ++i) {
+                lines.emplace_back(offset + i);
+            }
+        }
+        return true;
+    }
+
+    bool standard_syrec_synthesizer::unget_variables(const applications::variable_access::ptr& var, std::vector<unsigned>& lines [[maybe_unused]]) {
+        if (!var->indexes().empty()) {
+            // change offset
+
+            // check if it is not all numeric_expressions
+            unsigned n = var->var()->dimensions().size(); // dimensions
+            if ((unsigned)std::count_if(var->indexes().cbegin(), var->indexes().cend(), [&](const auto& p) { return dynamic_cast<applications::numeric_expression*>(p.get()); }) != n) {
+            }
+        }
+
+        return true;
+    }
```

Comment:
this function does nothing. is this an error or intended?
if it is an error, this should be fixed.
if not, this function can be removed.